README.md

```diff
@@ -4,121 +4,389 @@
+   section. In this configuration, Set the following config and add the claims you 
    need to retrieve (ex: http://wso2.org/claims/lastname) from the web app.
    
        Select Claim mapping Dialect - Use Local Claim Dialect
        
    See the example claim config below.
    ![Claim Config](https://user-images.githubusercontent.com/15249242/90488235-38d45580-e159-11ea-8beb-52d6b5c35034.png)
+   
+   vi. Click `Update` to save.
 
-       
-5. Try out the application by accessing the `http://localhost:8080/SampleApp/index.html`.
+#### Configuring The Webapp
+1. Download the `SampleApp.war` from the [latest release](https://github.com/asgardio/asgardio-tomcat-saml-agent/releases/latest).
+2. Deploy the application, `SampleApp.war` using Apache Tomcat.
+3. Try out the application by accessing the `http://localhost.com:8080/SampleApp/index.html`.
 
-![Recordit GIF](http://g.recordit.co/IvrtWnDnZ8.gif)
+![Recordit GIF](http://g.recordit.co/IvrtWnDnZ8.gif)    
 
-**NOTE:** Some browsers do not support cookie creation for naked host names (ex:- localhost). SSO functionality
- require cookies 
-in the browser. 
+## How it works
+### Classify secure resources, unsecured resources
+In the SampleApp sample, we have two pages. A landing page (`index.html`) which we have not secured, and another 
+page (`home.jsp`) which we have secured.
 
-In that case, use `localhost.com` host name for the sample application. You will require to edit the SampleApp
-.properties file in <TOMCAT_HOME>/webapps/SampleApp/WEB-INF/classes directory and set the following:
+`IndexPage` property of the sampleApp.properties file in the `<APP_HOME>/WEB-INF/classes` directory is used to define 
+the landing page of the webapp which is also considered as an unsecured page.
+Also the same page is used as the page that user get redirected once the logout is done.
+Here we have set `<APP_HOME>/index.html` as the value of `IndexPage` property.
+    IndexPage=/SampleApp/index.html
 
-`SAML2.AssertionConsumerURL=http://localhost.com:8080/SampleApp/home.jsp`
+By default, all other pages considered as secured pages. Hence `home.jsp` will be secured without any other configurations.
 
-and update the Assertion Consumer URLs in the Identity Server Service Provider configurations accordingly.
+### Trigger authentication
+In the `<APP_HOME>/index.html` page, we have added the action for the login button to trigger a SAML authentication:
+```
+<form action="samlsso?SAML2.HTTPBinding=HTTP-POST" method="post">
+    <div class="element-padding">
+        <input style="height: 30px; width: 60px" type="submit" value="log in">
+    </div>
+</form>
+```
 
-You will also require to add this entry 
-to `hosts` file. For windows this file locations is at `<Windows-Installation-Drive>\Windows\System32\drivers\etc
-\hosts`.
-For Linux/Mac OS, this file location is at `/etc/hosts`.
-## Installing the SDK
+This would trigger an authentication request, and redirect the user to the IdP authentication page.
```

Comment:
This **button click** would trigger an authentication...

Reply:
Resolved in https://github.com/asgardio/asgardio-tomcat-saml-agent/pull/1/commits/f25a1df8be1b219b41029eaf1195b4c5ddb16da4

Reply:
Resolved in https://github.com/asgardio/asgardio-tomcat-saml-agent/pull/1/commits/3135e1b3dc8a2ca466299cc3b94d6f3b8848af98